packages/spanner/src/database.js

```diff
@@ -233,6 +244,12 @@ Database.formatName_ = function(instanceName, name) {
  */
 Database.prototype.close = function(callback) {
   var self = this;
+  var leaks = this.pool_.getSessionLeaks();
+
+  if (leaks.length) {
+    callback(new SessionLeakError(leaks));
+    return;
```

Comment:
When `database.close` is called can we release all the sessions to the pool ourselves irrespective of whether they are being used or not so that the pooling library can be drained?

I agree it would be a good idea to investigate a different pooling library or get this pooling library's owner to add these features.